File: No_1288/Remove Covered Intervals.cpp
class Solution {
public:
    int removeCoveredIntervals(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end(), compare);
        int n_remove=0;
        for(int i=0; i<intervals.size();) {
            if(i == intervals.size()) break;
            int j = i+1;
            while(j<intervals.size() && intervals[i][1] >= intervals[j][1]) {
                n_remove++;
                j++;
            }
            i = j;
        }
        
        return intervals.size()-n_remove;
    }
    
    static bool compare(vector<int> a, vector<int> b){
        if(a[0] < b[0]) {
            return true;
        } else if(a[0] == b[0] && a[1] >= b[1]) {
            return true;
        }
        
        return false;
    }
};

File: No_1288/Remove Covered Intervals.py
class Solution:
    def removeCoveredIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort(key=lambda x: (x[0], -x[1]))
        
        ret = far = 0
        for _, e in intervals:
            if e > far:
                ret += 1
                far = e
        
        return ret